File: scripts/examine-csv.js
const fs = require('fs');
const csv = require('csv-parser');

console.log('Examining products.csv structure...');

const results = [];
let rowCount = 0;

fs.createReadStream('ecommerce-dataset/archive/archive/products.csv')
  .pipe(csv())
  .on('data', (data) => {
    if (rowCount < 5) {
      results.push(data);
    }
    rowCount++;
  })
  .on('end', () => {
    console.log('CSV Structure:');
    if (results.length > 0) {
      console.log('Columns:', Object.keys(results[0]));
      console.log('\nFirst 5 rows:');
      results.forEach((row, index) => {
        console.log(`Row ${index + 1}:`, row);
      });
    }
    console.log(`\nTotal rows in file: ${rowCount}`);
  }); 

File: examine_csv.py
import pandas as pd

# Read the first few rows of the CSV to understand the structure
df = pd.read_csv('ecommerce-dataset/archive/archive/products.csv', nrows=5)
print("CSV Structure:")
print(df.columns.tolist())
print("\nFirst 5 rows:")
print(df)
print(f"\nTotal rows in file: {len(pd.read_csv('ecommerce-dataset/archive/archive/products.csv'))}") 